shared/js/background/classes/ad-click-attribution-policy.js

```diff
@@ -173,4 +198,50 @@ export class AdClick {
         if (tab.site.baseDomain !== this.adBaseDomain) return false
         return this.hasNotExpired()
     }
+
+}
+
+export class AdClickDNR {
+    constructor (tabId, allowlist) {
+    this.allowlist = allowlist
+    this.tabId = tabId
+    this.initiatorDomain = null
+    this.rule  = generateDNRRule({
+            "id" : tabId,
+            "priority": CEILING_PRIORITY,
+            "actionType": "allow",
+            "requestDomains": allowlist.reduce((lst, entry) => { lst.push(entry.host); return lst}, []),
+        })
+    this.rule.condition.tabIds = [tabId]
+    }
+    
+    /**
+     * Create initial tab scoped DNR not limited to an initiatorDomain
+     * @param {Tab} tab
+     * @returns {integer}
+     **/
+    createInitialAdClickDNR () {
+        chrome.declarativeNetRequest.updateSessionRules({addRules: [this.rule]})
+    }
+
+    /*
+     * Update the tab DNR with initiator domain
+     */
+    addAdClickDNRInitiatorDomain (domain) {
+        this.initiatorDomain = domain
+        this.rule.condition.initiatorDomain = [domain]
+
+    }
+
+    removeAdClickDNR () {
```

Comment:
This needs to be triggered when the tab is navigated via the top URL.

I think we have similar clean up code in tab manager (or calls into it) I think we want to ensure we clear this out whenever a tab is removed etc.